test/parallel/test-child-process-validate-stdio.js

```diff
@@ -30,6 +30,19 @@ common.expectsError(() => _validateStdio(stdio2, true),
                     { code: 'ERR_IPC_SYNC_FORK', type: Error }
 );
 
+// should throw if stdio is not a valid input
+{
+  const stdio = ['foo'];
+  common.expectsError(() => _validateStdio(stdio, false),
+                      { code: 'ERR_INVALID_SYNC_FORK_INPUT', type: TypeError }
+  );
+}
+
+// should throw if stdio is not a valid option
+{
+  const stdio = [{ foo: 'bar' }];
+  assert.throws(() => _validateStdio(stdio), expectedError);
```

Comment:
Please do not mix `assert.throws()` and `common.expectsError()`. It will work but it is unnecessary overhead (I did see that you just used the existing pattern so this is not blocking). Instead, either only use `common.expectsError()` with a predefined error object such as:

```js
const expectedError = { code: 'ERR_INVALID_OPT_VALUE', type: TypeError };
common.expectsError(() => _validateStdio(stdio), expectedError);
```
Or
```js
const expectedError = {
  code: 'ERR_INVALID_OPT_VALUE',
  name: 'TypeError [ERR_INVALID_OPT_VALUE]'
};
assert.throws(() => _validateStdio(stdio), expectedError);
```

Reply:
Wow, this is a great review, let see if I understood; I should not mix `assert.throws` with `common.expectsError`, so I'll replace line 44 to this: `common.expectsError(() => _validateStdio(stdio), expectedError);` (?) I that way both methods are not mixed up

Reply:
Without changing `expectedError` this would not improve the situation.

An easy fix would be to change line line 9 as suggested above.